### PR TITLE
Fix banner crashing client when copying blank vanilia banner and desync in multiplayer

### PR DESCRIPTION
**Pull request by cylian914**
# Changes proposed in this pull request
- Fix banner crashing client on banner that didn't have tags (aka blank patterns)
- Fix server not copying but client did, causing a desync
- add a explanation on how to copy banner's patterns

## Testing
- [X] Yes I tested this before submitting it.
- [X] I also did a multiplayer test.

Review please

### CODE REVIEW

File: src/main/resources/assets/minecolonies/lang/manual_en_us.json [hunk 1]
```diff
@@ -1921,6 +1921,7 @@
   "com.minecolonies.coremod.gui.flag.base_layer": "Base",
   "com.minecolonies.coremod.gui.townhall.edit_flag": "Edit Colony Flag",
   "block.minecolonies.colony_banner": "Colony Flag Banner",
+  "com.mincolonies.coremod.item.colony_baner.tooltipempty": "Right click to copy pattern on the item",
```
Comment (Thodor12) [inline src/main/resources/assets/minecolonies/lang/manual_en_us.json]: Typos*

File: src/main/java/com/minecolonies/core/tileentities/TileEntityColonyFlag.java [hunk 2]
```diff
@@ -87,27 +82,29 @@ public void onDataPacket(final Connection net, final ClientboundBlockEntityDataP
...
+                this.patterns = colony.getColonyFlag();
                 setChanged();
             }
         }
 
-        return BannerBlockEntity.createPatterns(
-                DyeColor.WHITE,
-                this.flag.size() > 1 ? this.flag : this.patterns
+        List<Pair<Holder<BannerPattern>, DyeColor>> pattern = BannerBlockEntity.createPatterns(
+                DyeColor.WHITE, // <-- add a useless white base layer
```
Comment (Thodor12) [inline src/main/java/com/minecolonies/core/tileentities/TileEntityColonyFlag.java]: Please remove the unnecessary comments

File: src/main/java/com/minecolonies/core/items/ItemColonyFlagBanner.java [hunk 3]
```diff
@@ -46,49 +48,35 @@ public InteractionResult useOn(UseOnContext context)
...
 
     @Override
     public void appendHoverText(ItemStack stack, @Nullable Level worldIn, List<Component> tooltip, TooltipFlag flagIn)
     {
-        super.appendHoverText(stack, worldIn, tooltip, flagIn);
-
-        // Remove the base, as they have no translations (Mojang were lazy. Or maybe saving space?)
-        if (tooltip.size() > 1) tooltip.remove(1);
+	CompoundTag tag = BlockItem.getBlockEntityData(stack);
+	if (tag != null && tag.contains(TAG_BANNER_PATTERNS))
```
Comment (Thodor12) [inline src/main/java/com/minecolonies/core/items/ItemColonyFlagBanner.java]: Please follow the style guide (import the Minecolonies.xml into IntelliJ) or any of the other files for other editors.

File: src/main/java/com/minecolonies/core/items/ItemColonyFlagBanner.java [hunk 4]
```diff
@@ -46,49 +48,35 @@ public InteractionResult useOn(UseOnContext context)
...
-            }
-
-            CompoundTag tag = stack.getOrCreateTagElement("BlockEntityTag");
-            tag.put(TAG_BANNER_PATTERNS, patternList);
-
-            return InteractionResult.SUCCESS;
+	    BannerPattern.Builder patternsBuilder = new BannerPattern.Builder();
+	    List<Pair<Holder<BannerPattern>, DyeColor>> source;
+	    
+	    if (te instanceof BannerBlockEntity)
```
Comment (Thodor12) [inline src/main/java/com/minecolonies/core/items/ItemColonyFlagBanner.java]: Style guide

**Review by Thodor12 — CHANGES_REQUESTED** [review]

File: src/main/resources/assets/minecolonies/lang/manual_en_us.json (hunk 1 above)
Comment (cylian914) [reply src/main/resources/assets/minecolonies/lang/manual_en_us.json]: I'm dyslexic and don't see the typo, can you tell me whats wrong ?

Comment (MotionlessTrain) [reply src/main/resources/assets/minecolonies/lang/manual_en_us.json]: baner-> banner

Comment (MotionlessTrain) [reply src/main/resources/assets/minecolonies/lang/manual_en_us.json]: And mincolonies should probably be minecolonies

Comment (cylian914) [reply src/main/resources/assets/minecolonies/lang/manual_en_us.json]: thx

File: src/main/java/com/minecolonies/core/items/ItemColonyFlagBanner.java [hunk 5]
```diff
@@ -46,49 +48,36 @@ public InteractionResult useOn(UseOnContext context)
...
     }
 
     @Override
-    public void appendHoverText(ItemStack stack, @Nullable Level worldIn, List<Component> tooltip, TooltipFlag flagIn)
-    {
-        super.appendHoverText(stack, worldIn, tooltip, flagIn);
-
-        // Remove the base, as they have no translations (Mojang were lazy. Or maybe saving space?)
-        if (tooltip.size() > 1) tooltip.remove(1);
+    public void appendHoverText(ItemStack stack, @Nullable Level worldIn, List<Component> tooltip, TooltipFlag flagIn) {
```
Comment (Thodor12) [inline src/main/java/com/minecolonies/core/items/ItemColonyFlagBanner.java]: This is still not following the style guide, curly braces are supposed to be on the next line. Please format the code accordingly.

**Review by Thodor12 — CHANGES_REQUESTED** [review]

File: src/main/java/com/minecolonies/core/items/ItemColonyFlagBanner.java [hunk 6]
```diff
@@ -46,49 +48,42 @@ public InteractionResult useOn(UseOnContext context)
...
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/core/items/ItemColonyFlagBanner.java]: Move the curly to the next line

**Review by Thodor12 — DISMISSED** [review]

**Review by Raycoms — COMMENTED** [review]
Thanks for the submission. A small formatting thing is missing still. And an important question: Did you test this in singleplayer/multiplayer?

**Comment by cylian914** [comment]
yes i did tested it in multiplayer and singleplayer

**Review by Raycoms — APPROVED** [review]

**Review by someaddons — APPROVED** [review]

**Pull request merged**